crates/oxc_semantic/src/builder.rs

```diff
@@ -1078,9 +1078,9 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
         /* cfg - bb after if statement joins consequent and alternate */
         let after_if_graph_ix = self.cfg.new_basic_block();
 
-        if stmt.alternate.is_some() {
-            self.cfg.put_unreachable();
-        }
+        // if stmt.alternate.is_some() {
```

Comment:
For the code below, the return statement be treated as unreachable, would you like to give me some advice to fix this? @rzvxa 

```js
class Hello extends React.Component {
  render() {
    if (this.a > this.b) {
      foo1()
    } else {
      foo2()
    }

    return 'div'
  }
}
```

![CleanShot 2024-05-25 at 14.55.28@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/nUl2MXIwjUWxxQRgpP2H/617ee14c-abe2-4b71-9232-ae705cbbc226.png)

Reply:
I was considering removing the unreachable instructions so if it was a new rule I would've suggested waiting until the end of this weekend for me to finish my PR, But since there is an issue out in the wild let me take a closer look and see what workaround are possible.

The unreachable is redundant here, We should be able to remove it once memorizing the already visited nodes. You should also do the same treatment to the conditional expressions. Let me know if it breaks the `rules_of_hooks` since that's the only other rule doing extensive checks on conditional flows.

Reply:
I removed the redundant return instructions in if statement and condition statement, there are a bunch of cfg's snapshot updated, I think it's ok since you will refactor it soon :)

 > Let me know if it breaks the rules_of_hooks since that's the only other rule doing extensive checks on conditional flows.
 
 Review it again, find it safe for this rule in my view.